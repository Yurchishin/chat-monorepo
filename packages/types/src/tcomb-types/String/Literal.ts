import t, { Irreducible } from 'tcomb-validation'
import R from '@chat/ramda'

export const Literal = <T>(value: T, name?: string): Irreducible<T> => t.irreducible(
    name || 'Literal',
    R.equals(value),
)
