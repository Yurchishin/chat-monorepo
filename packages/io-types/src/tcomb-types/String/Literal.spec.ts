import {Literal} from './Literal'
import {expectTypeSuccess, expectTypeFailure} from "../../tests"

describe('Literal(\'literal\')', () => {
	const literal = Literal('literal')

	expectTypeSuccess(literal, 'literal')

	const literal2 = Literal('literal2', 'Name')
	expectTypeFailure(literal2, 'other')
	expectTypeFailure(literal2, 1)
})
