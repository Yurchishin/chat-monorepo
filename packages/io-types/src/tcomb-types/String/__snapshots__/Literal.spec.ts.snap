// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Literal('literal') expectTypeFailure(1): TYPE_FAILURE::Name(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Invalid value 1 supplied to Name",
      "path": Array [],
    },
  ],
  "value": 1,
}
`;

exports[`Literal('literal') expectTypeFailure(other): TYPE_FAILURE::Name(other) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "other",
      "expected": [Function],
      "message": "Invalid value \\"other\\" supplied to Name",
      "path": Array [],
    },
  ],
  "value": "other",
}
`;
